app/src/main/java/com/exallium/mvvmcleanexample/app/users/UserEditActivity.kt

```diff
@@ -14,45 +14,57 @@ import com.exallium.mvvmcleanexample.app.users.di.DaggerUserEditComponent
 import com.exallium.mvvmcleanexample.app.users.di.UserEditModule
 import com.exallium.mvvmcleanexample.databinding.UserEditViewBinding
 import com.exallium.mvvmcleanexample.presentation.users.UserEditViewModel
-import io.reactivex.disposables.CompositeDisposable
+import com.exallium.mvvmcleanexample.presentation.utils.autoEcoBinding
+import com.exallium.mvvmcleanexample.presentation.utils.ecoDisposable
 import javax.inject.Inject
 
-class UserEditView : AppCompatActivity() {
+class UserEditActivity : AppCompatActivity() {
 
     @Inject
     lateinit var viewModel: UserEditViewModel
 
     @Inject
     lateinit var router: AppRouter
 
-    val compositeDisposable = CompositeDisposable()
+    private var binding: UserEditViewBinding? by autoEcoBinding<UserEditViewBinding>(null)
+    private var disposable by ecoDisposable(null)
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        val binding = DataBindingUtil.setContentView<UserEditViewBinding>(this, R.layout.user_edit_view)
+        binding = DataBindingUtil.setContentView<UserEditViewBinding>(this, R.layout.user_edit_view).also { binding ->
+            DaggerUserEditComponent.builder()
+                    .appComponent(App.component)
+                    .domainUserModule(DomainUserModule())
+                    .userEditModule(UserEditModule())
+                    .build()
+                    .inject(this)
 
-        DaggerUserEditComponent.builder()
-                .appComponent(App.component)
-                .domainUserModule(DomainUserModule())
-                .userEditModule(UserEditModule())
-                .build()
-                .inject(this)
+            binding.viewModel = viewModel
 
-        binding.viewModel = viewModel
-
-        router.transitions().subscribe {
-            when (it) {
-                is AppRouter.Transition.GoBack -> onBackPressed()
-                is AppRouter.Transition.DisplayMessage -> displayMessage(it.message)
-                is AppRouter.Transition.DisplayError -> displayError(binding.root, it.message)
+            router.transitions().subscribe {
```

Comment:
if we're adding this extra step of scope, let's refactor some of the stuff in this method out into their own methods (like an inject method and a setupRouter method)